auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,137 @@
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (NullRawAuthHeaderException e) {
+            updatedRequest = request;
+        } catch (AuthHeaderNotBasicAuthException e) {
+            log.warn("AuthHeaderNotBasicAuthException", e);
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpRequest = (HttpServletRequest) request;
+            final String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return getRequestWithTokenFromRawAuthHeader(rawAuthHeader, httpRequest);
+        } else {
+            throw new AuthHeaderNotBasicAuthException("Request is not an HttpServletRequest.");
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromRawAuthHeader(String rawAuthHeader, HttpServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (rawAuthHeader == null) {
+            throw new NullRawAuthHeaderException("Raw auth header is null.");
+        } else if (isBasicAuth(rawAuthHeader)) {
+            final AuthHeader authHeader = getAuthHeader(rawAuthHeader);
+            return getRequestWithTokenFromAuthHeader(authHeader, request);
+        } else {
+            throw new AuthHeaderNotBasicAuthException("Auth header is not basic auth.");
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromAuthHeader(final AuthHeader authHeader, HttpServletRequest request) {
+        return new HttpServletRequestWrapper(request) {
+            @Override
+            public String getHeader(String name) {
+                if (Objects.equals(name, HttpHeaders.AUTHORIZATION)) {
+                    return authHeader.toString();
+                } else {
+                    return super.getHeader(name);
+                }
+            }
+        };
+    }
+
+    private AuthHeader getAuthHeader(String rawAuthHeader) {
+        final String password = getPassword(rawAuthHeader);
+        return AuthHeader.valueOf(password);
+    }
+
+    private String getPassword(String rawAuthHeader) {
+        String base64Credentials = rawAuthHeader.substring(BASIC_AUTH_STR.length()).trim();
+        String credentials;
+        try {
+            credentials = new String(BASE_64_ENCODING.decode(base64Credentials), StandardCharsets.UTF_8);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Could not decode credentials from auth header: " + e.getMessage());
+        }
+        return credentials.split(":", 2)[1];
```

Comment:
we should make sure there's actually a colon in the string and throw an exception if not -- otherwise this could result in a nasty `IndexOutOfBoundsException`